Firmware/SpoolJoin.cpp

```diff
@@ -0,0 +1,80 @@
+#include "SpoolJoin.h"
+#include "Marlin.h"
+#include "eeprom.h"
+
+namespace SpoolJoin {
+
+SpoolJoin spooljoin;
+
+SpoolJoin::SpoolJoin()
+    : status(EEPROM::Unknown)
+    , currentMMUSlot(0)
+{
+}
+
+void SpoolJoin::updateSpoolJoinStatus(EEPROM newStatus)
+{
+    status = newStatus;
+    eeprom_write_byte((uint8_t*)EEPROM_AUTO_DEPLETE, (uint8_t)status);
+}
+
+void SpoolJoin::initSpoolJoinStatus()
+{
+    EEPROM currentStatus = (EEPROM)eeprom_read_byte((uint8_t*)EEPROM_AUTO_DEPLETE);
+    if( currentStatus == EEPROM::Empty)
+    {
+        // By default SpoolJoin is disabled
+        updateSpoolJoinStatus(EEPROM::Disabled);
+    } else {
+        updateSpoolJoinStatus(currentStatus);
+    }
+
+    // Useful information to see during bootup
+    SERIAL_ECHOPGM("SpoolJoin is ");
+    if (isSpoolJoinEnabled())
+    {
+        SERIAL_ECHOLNPGM("enabled");
```

Comment:
:thinking: don't we have some already existing message (messages.h) which could be used instead of these "enabled/disabled" texts?

Reply:
We have "Off" and "On" those can serve the same purpose.

Reply:
Done. Saves 16 bytes. I used `_O()` on the strings since we don't care about translations in this case.